### DISCUSSION TITLE
translate_phrases.py  local variable 'audio_path' referenced before assignment

### DISCUSSION THREAD

**Discussion by fernandolamp** [cktlco/crew-chief-autovoicepack#38] [Troubleshooting] [answered]
Hi, i'm trying to use the translate the script but without success, can you help me?

```
#  python3 translate_phrases.py --target_language Portuguese --translated_phrase_inventory phrase_inventory_de.csv
Traceback (most recent call last):
  File "/app/translate_phrases.py", line 183, in <module>
    main()
  File "/app/translate_phrases.py", line 103, in main
    entries: List[CrewChiefAudioFile] = parse_phrase_inventory(
  File "/app/utils.py", line 59, in parse_phrase_inventory
    CrewChiefAudioFile(audio_path, audio_filename, subtitle, text_for_tts)
UnboundLocalError: local variable 'audio_path' referenced before assignment
```

i'm missing something?

tks!

**Comment by cktlco** [comment, marked as answer]
Hi! Excellent to see that you've started working on a Portuguese translation!

I can help for sure. I just fixed the bug you encountered and added some much better instructions. This approach is set up to use a local Ollama LLM instance, and there are few minor challenges around accessing that network service from the Docker container.

Improved instructions at top of script:
https://github.com/cktlco/crew-chief-autovoicepack/blob/main/translate_phrases.py#L12

Let me know if you see issues and we'll get it working! If you get really stuck, I can probably help generate the auto-translated phrases CSV file.

**Reply by cktlco** [reply]
You'll need to re-pull the docker image to pick up the changes (anytime after the [build](https://github.com/cktlco/crew-chief-autovoicepack/actions/runs/12335830754/job/34427707787) finishes):
`docker pull ghcr.io/cktlco/crew-chief-autovoicepack:latest`

**Reply by fernandolamp** [reply]
Works like a charm:

```
crew-chief-autovoicepack > python3 translate_phrases.py --target_language Portuguese --translated_phrase_inventory translated/phrase_inventory_pt.csv
INFO:root:Translation 1: okay -> tá bom (is good)

INFO:root:Translation 2: acknowledged -> reconhecido (recognized)

INFO:root:Translation 3: acknowledged -> reconhecido (recognized)

INFO:root:Translation 4: understood -> entendido (understood)

INFO:root:Translation 5: acknowledged -> reconhecido (recognized)

INFO:root:Translation 6: acknowledged -> reconhecido (recognized)

INFO:root:Translation 7: acknowledged -> reconhecido (recognized)
```

thanks!
